lib/ansible/modules/system/mount2.py

```diff
@@ -0,0 +1,410 @@
+#!/usr/bin/python
+# -*- coding: utf-8 -*-
+
+# (c) 2012, Red Hat, inc
+# Written by Seth Vidal
+# based on the mount modules from salt and puppet
+#
+# This file is part of Ansible
+#
+# Ansible is free software: you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation, either version 3 of the License, or
+# (at your option) any later version.
+#
+# Ansible is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with Ansible.  If not, see <http://www.gnu.org/licenses/>.
+
+
+ANSIBLE_METADATA = {'metadata_version': '1.0',
+                    'status': ['preview'],
+                    'supported_by': 'core'}
+
+
+DOCUMENTATION = '''
+---
+module: mount2
+short_description: Control active mount points
+description:
+  - This module controls active mount points.
+author:
+  - Ansible Core Team
+  - Seth Vidal
+  - Jonathan Mainguy
+version_added: "2.4"
+options:
+  path:
+    description:
+      - Path to the mount point (e.g. C(/mnt/files)).
+    required: true
+  src:
+    description:
+      - Device to be mounted on I(path). Required when I(state) set to C(mounted).
+    required: false
+    default: null
+  fstype:
+    description:
+      - Filesystem type. Required when I(state) is C(mounted).
+    required: false
+    default: null
```

Comment:
I would default to 'auto'  and not require it, let the command fail if it is not present in fstab (like mount does)